internal/cmd/skupper/token/nonkube/token_redeem.go

```diff
@@ -1,30 +1,96 @@
 package nonkube
 
 import (
+	"errors"
 	"fmt"
+	"github.com/skupperproject/skupper/internal/nonkube/client/fs"
+	"github.com/skupperproject/skupper/internal/utils/validator"
+	"github.com/skupperproject/skupper/pkg/apis/skupper/v2alpha1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"os"
 
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
+	"github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/scheme"
 	"github.com/spf13/cobra"
 )
 
 type CmdTokenRedeem struct {
-	CobraCmd  *cobra.Command
-	Flags     *common.CommandTokenRedeemFlags
-	Namespace string
-	siteName  string
+	CobraCmd           *cobra.Command
+	Flags              *common.CommandTokenRedeemFlags
+	Namespace          string
+	siteName           string
+	tokenHandler       *fs.TokenHandler
+	accessTokenHandler *fs.AccessTokenHandler
+	fileName           string
+	name               string
 }
 
 func NewCmdTokenRedeem() *CmdTokenRedeem {
 	return &CmdTokenRedeem{}
 }
 
 func (cmd *CmdTokenRedeem) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.Namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.tokenHandler = fs.NewTokenHandler(cmd.Namespace)
+	cmd.accessTokenHandler = fs.NewAccessTokenHandler(cmd.Namespace)
 }
 
-func (cmd *CmdTokenRedeem) ValidateInput(args []string) error { return nil }
-func (cmd *CmdTokenRedeem) InputToOptions()                   {}
+func (cmd *CmdTokenRedeem) ValidateInput(args []string) error {
+	var validationErrors []error
+	tokenStringValidator := validator.NewFilePathStringValidator()
+
+	// Validate token file name
+	if len(args) < 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("token file name must be configured"))
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("file name must not be empty"))
+	} else {
+		ok, err := tokenStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("token file name is not valid: %s", err))
+		} else {
+			cmd.fileName = args[0]
+		}
+	}
+
+	// Validate if a token file exists
+	if cmd.fileName != "" {
+		_, err := os.Stat(cmd.fileName)
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("token file does not exist: %s", err))
```

Comment:
Suggest "cannot open token file" or something to that effect. "does not exist" is not strictly correct.

Reply:
fixed